Observer/Order/BeforePlaceObserver.php

```diff
@@ -72,6 +64,39 @@ public function execute(Observer $observer): void
         }
         $publicOrderId = $this->checkoutSession->getBoldCheckoutData()['data']['public_order_id'] ?? '';
         $websiteId = (int)$order->getStore()->getWebsiteId();
-        $this->authorize->execute($publicOrderId, $websiteId);
+        $response = $this->authorize->execute($publicOrderId, $websiteId);
+        $this->saveTransaction($order, $response['data'] ?? []);
+    }
+
+    /**
+     * Add Bold transaction data to order payment.
+     *
+     * @param OrderInterface $order
+     * @param array $transactionData
+     * @return void
+     */
+    private function saveTransaction(OrderInterface $order, array $transactionData)
+    {
+        if (!isset($transactionData['transactions'][0]['transaction_id'])) {
+            return;
+        }
+        $order->getPayment()->setTransactionId($transactionData['transactions'][0]['transaction_id']);
+        $order->getPayment()->setIsTransactionClosed(0);
+        $order->getPayment()->addTransaction(Transaction::TYPE_AUTH);
+        $cardDetails = $transactionData['transactions'][0]['tender_details'] ?? null;
+        if (!$cardDetails) {
+            return;
+        }
+        $brand = $cardDetails['brand'] ?? '';
+        $lastFour = $cardDetails['last_four'] ?? '';
+        if (!$lastFour && isset($cardDetails['line_text'])) {
+            preg_match('/\b(\d{4})\b(?=\s*\(Transaction ID)/', $cardDetails['line_text'], $matches);
```

Comment:
[Q] wondering if we should also handle the case where the `line_text` is a single string with the last four digits. eg: 1111
The above works for PPCP due to how Checkout builds the line text, but for other payment gateways the line_text is simply the last four digits themselves.

Reply:
Done